Move plone.app.caching documentation to deployment/caching (#1449)
plone/documentation: docs/deployment/caching/profiles.md

## Custom caching profiles

Caching policies are often a compromise between speed and freshness.
More aggressive caching often comes at the cost of increased risk of stale responses.
The provided default profiles tend to err on the side of freshness over speed, leaving room for greater speed if desired.

Customization may also be needed if third-party products are installed which require special treatment.
Examine the HTTP response headers to determine whether the third-party product requires special treatment.
Most simple cases probably can be solved by adding the content type or template to the appropriate mapping.
More complicated cases may require custom caching operations.

A GenericSetup profile used for caching should be registered for the `ICacheProfiles` marker interface to distinguish it from more general profiles used to install a product.
This also hides the profile from Plone's {guilabel}`Add-ons` control panel.

Here is an example from this package:

```xml
<genericsetup:registerProfile
    name="with-caching-proxy"
    title="With caching proxy"

>     description="Settings useful for setups with a caching proxy such as Squid or Varnish"

Reviewer: ```suggestion
    description="Settings useful for setups with a caching proxy such as Varnish or a CDN"
```

---
myst:
  html_meta:
    "description": "Caching profiles for Plone"
    "property=og:description": "Caching profiles for Plone"
    "property=og:title": "Caching profiles for Plone"
    "keywords": "Plone, deployment, automation, caching"
---

(caching-profiles-label)=

# Caching profiles

All persistent configuration for the caching machinery is stored in the configuration registry, as managed by `plone.app.registry`.

This can be modified using the `registry.xml` GenericSetup import step.

The *Import settings* tab of the control panel allows you to import these caching profiles.

## Default caching profiles

> `plone.app.caching` includes three default caching profiles.
> 
> Two of these  profiles encapsulate the cache settings that are known to work well with a typical default Plone installation:

Reviewer: ```suggestion
`plone.app.caching` includes two default caching profiles.
These  profiles encapsulate the cache settings that are known to work well with a typical default Plone installation:
```

### Without caching proxy

Settings useful for setups without a caching proxy.

### With caching proxy

> Settings useful for setups with a caching proxy such as Squid or Varnish.

Reviewer: ```suggestion
Settings useful for setups with a caching proxy such Varnish or a CDN.
```

> The only difference from the "without caching proxy" profile are some settings to enable proxy caching of files/images in content space and content feeds.

Reviewer: I am pretty sure this is not true anymore.

Author: I just checked the profiles, and it is it. Feed and Files use `moderateCaching` with `smaxage`:

## With Proxy
```xml
  <!-- plone.content.file -->
  <record name="plone.app.caching.moderateCaching.plone.content.file.smaxage">
    <field ref="plone.app.caching.moderateCaching.smaxage" />
    <value>86400</value>
  </record>
  <record name="plone.app.caching.moderateCaching.plone.content.file.lastModified">
    <field ref="plone.app.caching.moderateCaching.lastModified" />
    <value>True</value>
  </record>
```

## Without Proxy
```xml
  <record name="plone.app.caching.weakCaching.plone.content.file.lastModified">
    <field ref="plone.app.caching.weakCaching.lastModified" />
    <value>True</value>
  </record>
```
